scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -151,3 +152,37 @@ def prepare_model(spec_path, model_config, registry_name, temp_dir):
     except Exception as e:
         logger.log_error(f"prepare model failed for {spec_path}. Error {e}")
         return None, False
+
+
+def update_model_metadata(
+        model_name: str,
+        model_version: str,
+        registry_name: str,
+        update: AssetVersionUpdate):
+    """Update the mutable metadata of already registered Model."""
+    try:
+        ml_client = MLClient(credential=AzureCliCredential(), registry_name=registry_name)
+        model = ml_client.models.get(name=model_name, version=model_version)
+
+        if update is not None:  # to update older version of models
+            if update.description is not None:
+                model.description = update.description
+            if update.tags:
+                # Replace tags
+                if update.tags.replace is not None:
+                    model.tags = update.tags.replace
+                elif update.tags.add or update.tags.delete:
+                    # Add tags
+                    if update.tags.add:
+                        for k, v in update.tags.add.items():
+                            model.tags[k] = v
+
+                    # Delete tags
+                    if update.tags.delete:
+                        for k in update.tags.delete:
+                            model.tags.pop(k, None)
+
+        ml_client.models.create_or_update(model)
```

Comment:
Consider identifying whether any changes were made to the model object, in which case an update wouldn't be necessary.

Reply:
Updated. Could you please help to review